### DISCUSSION TITLE
What is the network setup script (for nspawn containers) for?

### DISCUSSION THREAD

**Discussion by robigan** [unifi-utilities/unifi-common#617] [General]
When going through the setup of nspawn containers and getting a container running. Option 2A specifies that one can set up a MacVLAN to give the container its own MAC, etc. Now the docs say to download the script, configure it and then execute it, apart from its secondary functions for DNSmasq and forcing DNS to pass through the container, the script sets up a MacVLAN attached to the desired interface. Is this really necessary? The nspawn configuration provided sets `MACVLAN` to `br5`, not the `br5.mac` created. Additionally the systemd nspawn docs say that the `MACVLAN=` option simply indicates to nspawn which interface the MACVLAN it creates should be attached to.
> MACVLAN=, IPVLAN=
           Takes a space-separated list of interfaces to add MACLVAN or
           IPVLAN interfaces to, which are then added to the container.
           The interface object is defined either by a single interface
           name, referencing the name on the host, or a colon-separated
           pair of interfaces, in which case the first one references
           the name on the host, and the second one the name in the
           container. These options correspond to the --network-macvlan=
           and --network-ipvlan= command line switches and imply
           Private=yes. These options are privileged (see above).
> 
> Added in version 226.

The docs seem to imply that nspawn creates its own macvlan interface. I'm therefore a little confused, does the script really need to make its own MacVLAN?

**Comment by m4n63** [comment]
My understanding is that the script is needed as it creates the  host bridge (gw) and adds the (host)route for the container. nspawn is then bound to this vlan, and creates its own network interface inside it.

(but I am not considering myself to be an expert on this)
